File: ios-app/MobileApp/MobileApp/Models/APIError.swift
//
//  APIError.swift
//  MobileApp
//
//  Created on 2025-01-08.
//

import Foundation

struct APIError: Codable {
    let error: ErrorDetail
    let timestamp: String
}

struct ErrorDetail: Codable {
    let code: String
    let message: String
    let details: String?
}

enum NetworkError: Error {
    case invalidURL
    case noData
    case decodingError
    case serverError(String)
    case unauthorized
    case unknown
}

File: src/android-app/app/src/main/java/com/example/mobileapp/models/ApiError.java
package com.example.mobileapp.models;

public class ApiError {
    private ErrorDetail error;
    private String timestamp;

    public ErrorDetail getError() {
        return error;
    }

    public void setError(ErrorDetail error) {
        this.error = error;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}

class ErrorDetail {
    private String code;
    private String message;
    private String details;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }
}
